service/src/main/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiDispatch.java

```diff
@@ -169,4 +195,22 @@ private ApiAzureLandingZoneResult fetchCreateAzureLandingZoneResult(String jobId
         .errorReport(MapperUtils.ErrorReportMapper.from(jobResult.getApiErrorReport()))
         .landingZone(azureLandingZone);
   }
+
+  /*
+   * Note: This method is an initial implementation that must be revised,
+   * and likely be refactored once WSM stores the LZ id in the azure context.
+   * The initial assumption is that the cardinality of 1:1 between the cloud context and the LZ
+   * is enforced in the create operation, therefore more than one LZ per azure context is not allowed.
+   */
+  public String getLandingZoneId(AzureCloudContext azureCloudContext) {
+    return getFirstLandingZoneId(MapperUtils.LandingZoneTargetMapper.from(azureCloudContext))
+        .orElseThrow(
+            () ->
+                new IllegalStateException(
```

Comment:
In this particular method we are trying to find LZ. Wouldn't be LandingZoneNotFoundException more informative? Not sure if IllegalStateException is appropriate here.

Reply:
The cause of the exception is that the creation flight is not complete (the db is not updated), hence a timing issue in which case an IllegalStateException is appropriate. I will add more details to the error message to make this clearer.